From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sun, 4 Jun 2023 16:33:21 +0100
Subject: [PATCH] Ensure block->item material conversion

Placing an oak sign on a wall will not show on snitches because OAK_WALL_SIGN is not an item material. This defers this to a new function in MaterialUtils.

diff --git a/src/main/java/com/untamedears/jukealert/model/actions/abstr/LoggableBlockAction.java b/src/main/java/com/untamedears/jukealert/model/actions/abstr/LoggableBlockAction.java
index 9bf22e376ba99b48dd156409f19e2960c5caa4bb..62aa65759fa4f02a7c225c6b6ba7cfe07e292444 100644
--- a/src/main/java/com/untamedears/jukealert/model/actions/abstr/LoggableBlockAction.java
+++ b/src/main/java/com/untamedears/jukealert/model/actions/abstr/LoggableBlockAction.java
@@ -12,6 +12,11 @@ import vg.civcraft.mc.civmodcore.inventory.gui.DecorationStack;
 import vg.civcraft.mc.civmodcore.inventory.gui.IClickable;
 import vg.civcraft.mc.civmodcore.inventory.items.ItemUtils;
 
+// CivPatched Start
+import java.util.Objects;
+import vg.civcraft.mc.civmodcore.inventory.items.MaterialUtils;
+// CivPatched End
+
 public abstract class LoggableBlockAction extends LoggablePlayerAction {
 
 	protected final Location location;
@@ -39,16 +44,7 @@ public abstract class LoggableBlockAction extends LoggablePlayerAction {
 	the correct inventory item to represent.
 	 */
 	public Material getMaterial() {
-		if (material == Material.POWDER_SNOW) {
-			return Material.POWDER_SNOW_BUCKET;
-		}
-		if (material == Material.FIRE) {
-			return Material.FLINT_AND_STEEL;
-		}
-		if (material == Material.SWEET_BERRY_BUSH) {
-			return Material.SWEET_BERRIES;
-		}
-		return material;
+		return Objects.requireNonNullElse(MaterialUtils.getBlocksItemMaterial(this.material), this.material); // CivPatched
 	}
 
 	@Override
@@ -58,10 +54,14 @@ public abstract class LoggableBlockAction extends LoggablePlayerAction {
 
 	@Override
 	public IClickable getGUIRepresentation() {
-		ItemStack is;
-		try {
-			is = new ItemStack(getMaterial());
-		} catch (Exception e) {
+		// CivPatched Start
+		final Material material = getMaterial();
+		final ItemStack is;
+		if (ItemUtils.isValidItemMaterial(material)) {
+			is = new ItemStack(material);
+		}
+		else {
+			// CivPatched End
 			is = new ItemStack(Material.STONE);
 			ItemUtils.addLore(is,
 					String.format("%sMaterial: %s%s", ChatColor.GOLD, ChatColor.AQUA, ItemUtils.getItemName(getMaterial())));
